test/blackbox-tests/test-cases/pkg/solver-flags-per-context.t

```diff
@@ -0,0 +1,102 @@
+
+Helper shell function that generates an opam file for a package:
+  $ mkpkg() {
+  >   name=$1
+  >   mkdir -p mock-opam-repository/packages/$name/$name.0.0.1
+  >   cat >mock-opam-repository/packages/$name/$name.0.0.1/opam
+  > }
+
+Helper shell function to generate a dune-project file and generate lockdir:
+  $ solve_project() {
+  >   cat >dune-project
+  >   dune pkg lock --opam-env=pure --opam-repository=mock-opam-repository --all-contexts
+  > }
+
+Create a workspace file with some contexts with different combinations of with-test and with-doc flags
+  $ cat >dune-workspace <<EOF
+  > (lang dune 3.8)
+  > (context
+  >  (default))
+  > (context
+  >  (default
+  >   (name default-solver-env)
+  >   (lock default-solver-env.lock)))
+  > (context
+  >  (default
+  >   (name default-solver-flags)
+  >   (lock default-solver-flags.lock)
+  >   (solver_env)))
+  > (context
+  >  (default
+  >   (name empty-solver-flags)
+  >   (lock empty-solver-flags.lock)
+  >   (solver_env
+  >    (flags))))
+  > (context
+  >  (default
+  >   (name with-test-only)
+  >   (lock with-test-only.lock)
+  >   (solver_env
+  >    (flags with-test))))
+  > (context
+  >  (default
+  >   (name with-doc-only)
+  >   (lock with-doc-only.lock)
+  >   (solver_env
+  >    (flags with-doc))))
+  > (context
+  >  (default
+  >   (name with-doc-and-with-test)
+  >   (lock with-doc-and-with-test.lock)
+  >   (solver_env
+  >    (flags with-doc with-test))))
```

Comment:
I wonder whether it would make sense to implement the set language so `(flags :standard)` can be specified.

Reply:
Ah yes I like the idea of allowing `:standard` here. It won't be hard to add as long as we don't allow `:include` which I think is reasonable.

Reply:
Updated to use the ordered set language for this field and added a test using the `:standard` keyword.